### PR TITLE
State Fixes and content update

### PR DESCRIPTION
**Pull request by bkayser**
Fixed some issues with state management and user authentication.

Also removed all the dipshit stuff.

### CODE REVIEW

File: app/src/components/Avatar.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ function hash(letters: string): number {
 
 /*
  * Allow three different states.  If there is no contact, use an anonymous icon.
- * If there is no contact avatar, use the initials.  Otehrwise use the contact avatar
```
Comment (saxond) [inline app/src/components/Avatar.tsx:22]: lol

File: app/src/components/Avatar.tsx [hunk 2]
```diff
@@ -29,19 +29,22 @@ export default function Avatar({contact}: Props) {
...
+            letters = ((contact.first ? contact.first.slice(0,1) : "") +
+                (contact.last ? contact.last.slice(0,1) : "")).toUpperCase()
             const bgColors = ["navy", "pink", "red", "blue", "green"];
-            bg = bgColors[hash(contact.last) % bgColors.length]
+            bg = bgColors[hash(letters) % bgColors.length]
+        } else {
+            anonymous = true;
         }
     }
     let class_name = "avatar";
```
Comment (saxond) [inline app/src/components/Avatar.tsx:45]: I just noticed this..  Why `class_name` and not `className`?

File: app/src/index.tsx [hunk 3]
```diff
@@ -12,58 +12,67 @@ import {fetchUser, SessionExpired} from "./data";
...
+            removeCachedUserToken();
         }
     }, [error])
 
+    // Re-establish session
     useEffect(() => {
-        if (userToken) {
-            fetchUser(userToken["access_token"]).then((u: ResolvedUser) => {
-                // eslint-disable-next-line no-debugger
+        if (getCachedUserToken() && !error && !resolvedUser ) {
```
Comment (saxond) [inline app/src/index.tsx:42]: Two calls to `getCachedUserToken()`?  Wanna throw that in a `const` local variable?

File: app/src/index.tsx [hunk 4]
```diff
@@ -105,4 +114,6 @@ const root = createRoot(container);
 root.render(
     <GoogleOAuthProvider clientId={CLIENT_ID}>
         <Home/>
-    </GoogleOAuthProvider>);
\ No newline at end of file
+    </GoogleOAuthProvider>);
+
+export {ErrorContext}
```
Comment (saxond) [inline app/src/index.tsx:119]: 🤔

File: app/src/routes/contact.edit.tsx [hunk 5]
```diff
@@ -6,15 +6,15 @@ import {Link} from 'react-router-dom';
...
 
 interface Props {
     id?: string;
 }
 
 export default function EditContact({id}: Props) {
-    const [loading, setLoading] = useState(false);
-    const [contact, setContact] = useState<ContactRecord>(null);
+    const [loading, setLoading] = useState(true);
+    const [contact, setContact] = useState<ContactRecord>({id:null, createdAt:''});
```
Comment (saxond) [inline app/src/routes/contact.edit.tsx:17]: Why do you need to initialize an empty contact?

File: app/src/components/Avatar.tsx (hunk 2 above)
Comment (bkayser) [reply app/src/components/Avatar.tsx:45]: mistake.  fixed.

File: app/src/index.tsx (hunk 3 above)
Comment (bkayser) [reply app/src/index.tsx:42]: Sure.

File: app/src/index.tsx (hunk 4 above)
Comment (bkayser) [reply app/src/index.tsx:119]: That was how I saw it in the context demo.  I moved the export to the declaration.

File: app/src/routes/contact.edit.tsx (hunk 5 above)
Comment (bkayser) [reply app/src/routes/contact.edit.tsx:17]: I think I was chasing down a bug and grasping for straws.  Replaced with null.

**Pull request merged**